components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/ToolbarPresenter.kt

```diff
@@ -36,6 +36,7 @@ class ToolbarPresenter(
     internal var renderer = URLRenderer(toolbar, urlRenderConfiguration)
 
     private var scope: CoroutineScope? = null
+    private val toolbarActionSet = HashSet<WebExtensionToolbarAction>()
```

Comment:
As a web extension can only have one browser action or page action, I think we could use a `map` instead of a `HashSet`, where the key will be the `WebExtension.id` and the value will be the `WebExtensionToolbarAction` object. This way we can make sure that `WebExtensionToolbarAction`s are per `WebExtensionToolbarAction`.

Also this way you don't have to override `equals` and `hashcode` in `WebExtensionToolbarAction`.

Reply:
@Amejia481 Hmm, I think `HashSet` has constant lookup time. If we use map as you suggest, it'd be as good as using a list since we'd always iterate every time render() is fired up. Is there any other reason for using a map aside from eliminating duplicates?

Reply:
Unless I'm misunderstanding here, are we checking against `WebExtension.id`, not `BrowserAction` references?